src/Makefile

```diff
@@ -127,8 +135,22 @@ tools: libpmem libpmemblk libpmemlog libpmemobj
 libpmemblk libpmemlog libpmemobj: libpmem
 benchmarks test tools: common
 
-pkg-config:
-	@$(MAKE_PKG_CONFIG) "$(prefix)" "$(libdir)" "$(SRCVERSION)"
+# re-generate *.pc files if version, prefix or libdir changes
+old_pc=$(strip $(shell cat $(PKG_CONFIG_COMMON) 2>/dev/null))
+new_pc="version=$(SRCVERSION) libdir=$(libdir) prefix=$(prefix)"
+update_pc=$(findstring $(old_pc), $(new_pc))
+
+pkg-cfg-common:
+	@$(if $(update_pc),, echo -e "\nversion=$(SRCVERSION)\nlibdir=$(libdir)\nprefix=$(prefix)" > $(PKG_CONFIG_COMMON))
```

Comment:
Please drop \n from the beginning, add it at the end and drop initial newline from templates.